### PR TITLE
Display owners for relics & mythics (whenever they're implemented)

### PR DESCRIPTION
**Pull request by rude-jerk**
image

Displaying these on the character page is a task for a future date.

Input on styling would be great, I do prefer to avoid a single-column table, though.

### CODE REVIEW

**Comment by Mozzy4Ever** [comment]
Single coluuuuuuuuumn

**Comment by bkesecker** [comment]
A grid probably won't work well on mobile unless side scrollable.  This is a 'badge of honor' feature, so I am not sure of purpose beyond that to make it more user friendly.  Nice idea!

**Comment by bkesecker** [comment]
Working in test.

image

http://test.edenxi.org:8080/tools/item/amanomurakumo

File: api/v1/utils/items.js [hunk 1]
```diff
@@ -84,6 +86,23 @@ const getBazaars = async (query, itemname) => {
...
+const getOwners = async (query, itemname) => {
+  // Prevent any funny business
+  if (!owner.owner_item_list.includes(itemname)) {
+    return [];
+  }
+  try {
+    const statement = `SELECT DISTINCT charname FROM chars c 
+        JOIN char_inventory i  ON i.charid = c.charid 
+        JOIN item_basic b ON b.itemid = i.itemId 
+        WHERE b.name = ? ORDER BY charname ASC;`;
```
Comment (bkesecker) [inline api/v1/utils/items.js]: `WHERE b.name = ? AND c.deleted IS NULL` should help with performance a bit.

**Review by bkesecker — CHANGES_REQUESTED** [review]

**Comment by bkesecker** [comment]
Query took 16 seconds with the 40,000 characters in the DB.

**Comment by rude-jerk** [comment]
Easy enough! 

Did the query take 16 seconds without the suggested change or with it made locally? It's a bit hard to simulate that number of characters and inventories in the database, though I probably should write a script to fake it for the sake of testing locally.

**Comment by InoUno** [comment]
Since it's such expensive queries (having to search through *all* items on the server), you'll probably want to cache it quite heavily.

Maybe it's better to make the server keep an updated list for each of the items, and have it refresh that list every X hours (without user action). That way users won't have to wait 10+ seconds each time they load one of the relic item pages to see results.

**Comment by rude-jerk** [comment]
I can also remove one of the query joins by storing the itemids for the items to perform this with rather than the names, but didn't bother.  The DISTINCT keyword also adds a bit of overhead and can probably be pulled, though I included it because of a few hesitations since rare singularity isn't enforced by the DB, but rather by the application.

**Comment by rude-jerk** [comment]
@bkesecker Can you give this a try again? I made a few changes and locally I see a fairly large query performance difference, though I admit I'm only faking ~10,000 characters in my database.

**Comment by slassen** [comment]
@rude-jerk the chars table won’t be the bottleneck. It’ll be the char_inventory. I haven’t had access to the database in quite some time but you can safely assume maybe 50 inventory per character and we had maybe 40000 chars? So about 2 million items; most likely more.

**Comment by rude-jerk** [comment]
Yeah, I tested this with 1.4m records in my local database and removing the items join reduced the query time by ~half.

**Comment by slassen** [comment]
Cool. I will review this week.

**Review by InoUno — APPROVED** [review]

**Pull request merged**